.github/workflows/build_and_test.yaml

```diff
@@ -0,0 +1,76 @@
+name: Build and Test PR
+
+on:
+  pull_request: # Trigger for pull requests.
+    types: [opened, synchronize, reopened]
+  workflow_dispatch: # Allows for manual triggering.
+    inputs:
+      ref:
+        description: "The ref to build and test."
+        required: False
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.inputs.ref || github.ref }}
+
+      - name: Lint
+        shell: bash
+        run: build/check.py
+
+  build_and_test:
+    # Don't waste time doing a full matrix of test runs when there was an
+    # obvious linter error.
+    needs: lint
+    strategy:
+      matrix:
+        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
+        include:
+          - os: ubuntu-latest
+            # Use headless browsers on Linux.
+            browser: ["ChromeHeadless", "FirefoxHeadless", "EdgeHeadless"]
+            extra_flags: "--use-xvfb"
+          - os: macos-latest
+            browser: ["Firefox", "Chrome", "Safari"]
```

Comment:
Is it possible add Edge also?

Reply:
Well, according to `--browsers help`, the available browsers on my macOS machine are:
```
  Chrome
  ChromeHeadless
  ChromeCanary
  ChromeCanaryHeadless
  Firefox
  FirefoxHeadless
  Safari
```

According to https://github.com/karma-runner/karma-edge-launcher/issues/10, `@chiragrupani/karma-chromium-edge-launcher` (which we do load) apparently should support Edge macOS, but I'm not seeing it in my list, even after installing Edge locally.

Well, I can look into it as a follow-up, I suppose. I'm already going to be investigating the integration test failures in macOS as a follow-up.

Reply:
On my mac:

```
alvaro.velad@MBP-de-Alvaro shaka-player % ./build/test.py --quick
[WARNING] Using default browsers: ['Chrome', 'Edge', 'Firefox', 'Safari']
[INFO] Generating Closure dependencies...
[INFO] Compiling the library (ui, release)...
[INFO] Running test (1 / 1, 0 failed so far)...
(node:19552) Warning: Accessing non-existent property 'VERSION' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
Chrome 97.0.4692.71 (Mac OS 10.15.7): Executed 1947 of 1983 (skipped 36) SUCCESS (56.526 secs / 53.278 secs)
Firefox 95.0 (Mac OS 10.15): Executed 1901 of 1983 (skipped 82) SUCCESS (57.5 secs / 55.125 secs)
Edge 97.0.1072.55 (Mac OS 10.15.7): Executed 1901 of 1983 (skipped 82) SUCCESS (56.172 secs / 52.351 secs)
Safari 15.2 (Mac OS 10.15.7): Executed 1901 of 1983 (skipped 82) SUCCESS (30.098 secs / 26.149 secs)
TOTAL: 7650 SUCCESS
[INFO] Run complete
[INFO] Result (exit code): 0
alvaro.velad@MBP-de-Alvaro shaka-player % uname -amnprsv
Darwin MBP-de-Alvaro 20.6.0 Darwin Kernel Version 20.6.0: Mon Aug 30 06:12:21 PDT 2021; root:xnu-7195.141.6~3/RELEASE_X86_64 x86_64 i386
```

@theodab , so Edge in macOS is working